controllers/userDetail.js

```diff
@@ -1,18 +1,23 @@
 const User = require('../models/user');
 
 const detail = async (req, res) => {
-  const activeAcc = await User.findById(req.user).exec();
-  const careTradesTitle = 'Mijn eigenschappen:';
-  const oldTradesTitle = 'Wat de ouderen verwacht van jou:';
-  const user = await User.findById(req.params.userId).exec();
-  const title = user.name;
-  res.render('pages/user-detail', {
-    activeAcc,
-    user,
-    oldTradesTitle,
-    careTradesTitle,
-    title,
-  });
+  try {
+    const activeAcc = await User.findById(req.user).exec();
+    const careTradesTitle = 'Mijn eigenschappen:';
+    const oldTradesTitle = 'Wat de ouderen verwacht van jou:';
+    const user = await User.findById(req.params.userId).exec();
+    const title = user.name;
+
+    res.render('pages/user-detail', {
+      activeAcc,
+      user,
+      oldTradesTitle,
+      careTradesTitle,
+      title,
+    });
+  } catch (error) {
+    res.redirect('/');
+  }
```

Comment:
Goed dat het nu niet mogelijk is om een detailpagina op te zoeken van een id dat niet in de database staat.

Reply:
Alleen de const user&title moeten in de try/catch. De andere const's moeten boven try staan.

Reply:
gedaan!